server/api/routes.ts

```diff
@@ -188,17 +218,26 @@ router.get('/cla/getproject/:id', (req, res, next) => {
 
 // Delete CLA
 router.post('/cla/delete', async(req, res, next) => {
-  pack(claAPI.deleteCLA(req, req.body), res, next);
+  const access = (req as any).UserAccess;
+  if (access.includes(AccessTypes.admin)) {
+    pack(claAPI.deleteCLA(req, req.body), res, next);
+  };
 });
 
 // Edit CLA
 router.post('/cla/update', async (req, res, next) => {
-  pack(claAPI.updateCLA(req, req.body), res, next);
+  const access = (req as any).UserAccess;
+  if (access.includes(AccessTypes.admin)) {
+    pack(claAPI.updateCLA(req, req.body), res, next);
+  };
 });
 
 // New CLA post request
 router.post('/cla/submit', async (req, res, next) => {
-  pack(claAPI.addNewCLA(req, req.body), res, next);
+  const access = (req as any).UserAccess;
+  if (access.includes(AccessTypes.admin)) {
+    pack(claAPI.addNewCLA(req, req.body), res, next);
+  };
```

Comment:
These routes will fail in a weird way if your access check fails; instead of getting an error they'll just hang and the connection won't close. You'll want to handle that either here, or if you want by adding a second function/middleware.

Reply:
Handling here for now and will revisit if/when we upgrade to koa.